components/tools/OmeroPy/src/omero/plugins/metadata.py

```diff
@@ -0,0 +1,403 @@
+        om = (("Global", global_om),
+              ("Series", series_om))
+
+        for name, tuples in om:
+            # Matches the OMERO4 original_metadata.txt format
+            self.ctx.out("[%sMetadata]" % name)
+            for k, v in tuples:
+                self.ctx.out("%s=%s" % (k, v))
+
+    def _output_ann(self, mdobj, funcstr, parents, indent):
+        try:
+            anns = getattr(mdobj, funcstr)()
+        except NotImplementedError:
+            self.ctx.err('WARNING: Failed to get annotations for %s' %
+                         mdobj.get_name())
+            anns = []
+        if indent is not None:
+            self.ctx.out("%s%s" % (
+                '  ' * indent, mdobj.get_name()))
+            indent += 1
+        for a in anns:
+            self.ctx.out(self._format_ann(mdobj, a, indent))
+        if parents:
+            for p in mdobj.get_parents():
+                self._output_ann(p, funcstr, parents, indent)
+
+    def bulkanns(self, args):
+        ("Provide a list of the NSBULKANNOTATION tables linked "
+         "to the given object")
+        md = self._load(args)
+        indent = None
+        if args.report:
+            indent = 0
+        self._output_ann(md, 'get_bulkanns', args.parents, indent)
+
+    def measures(self, args):
+        ("Provide a list of the NSMEASUREMENT tables linked "
+         "to the given object")
+        md = self._load(args)
+        indent = None
+        if args.report:
+            indent = 0
+        self._output_ann(md, 'get_measures', args.parents, indent)
+
+    def mapanns(self, args):
+        "Provide a list of all MapAnnotations linked to the given object"
+        md = self._load(args)
+        mas = md.get_allanns()
+        for ma in mas:
+            if ma.get_type() == 'MapAnnotation':
+                if args.report:
+                    self.ctx.out(self._format_ann(md, ma, 0))
+                else:
+                    self.ctx.out(self._format_ann(md, ma))
+
+    def allanns(self, args):
+        "Provide a list of all annotations linked to the given object"
+        md = self._load(args)
+        for a in md.get_allanns():
+            if args.report:
+                self.ctx.out(self._format_ann(md, a, 0))
+            else:
+                self.ctx.out(self._format_ann(md, a))
+
+    # WRITE
+
+    def populate(self, args):
+        "Add metadata (bulk-annotations) to an object"
+        client = self.ctx.conn(args)
+        # TODO: Configure logging properly
+        if args.report:
+            populate_metadata.log.setLevel(logging.DEBUG)
+        else:
+            populate_metadata.log.setLevel(logging.INFO)
+        context_class = dict(self.POPULATE_CONTEXTS)[args.context]
+        ctx = context_class(client, args.obj, args.file)
+        ctx.parse()
+        if not args.dry_run:
+            ctx.write_to_omero()
+
+    def rois(self, args):
+        "Manage ROIs"
+        md = self._load(args)
+        if args.delete:
```

Comment:
Added for consistency, passed straight through to `omero delete`. In theory `omero rois Object:Id` should show the list of ROIs that would be deleted